Handle HTTP chunking across multi-byte boundaries

Freaky opened the pull request:
When a response contains multi-byte UTF-8 characters that cross chunk boundaries, appending the two halves to the intermediate string results in two Unicode Replacement Characters being inserted instead of the two halves being joined.

Use StringDecoder to buffer the partial characters across chunks.

Resolves #22

Commit:
Handle HTTP chunking across multi-byte boundaries

When a response contains multi-byte UTF-8 characters that cross chunk
boundaries, appending the two halves to the intermediate string results
in two Unicode Replacement Characters being inserted instead of the two
halves being joined.

Set the response encoding to UTF-8 to ensure it buffers partial
characters between chunks.

Resolves #22

Co-authored-by: zyc9012 <zyc9012@gmail.com>

zyc9012 commented on src/utils.ts:
```diff
@@ -63,15 +64,17 @@ export function execute(
   return new Promise((resolve, reject) => {
     let timer: number;
     const req = https.get(url, (resp) => {
+      const decoder = new StringDecoder("utf8");
```
Simply adding `resp.setEncoding("utf8");` here should do the job, what do you think?

[Documentation](https://nodejs.org/docs/latest-v7.x/api/stream.html#stream_readable_setencoding_encoding)

Freaky replied:
That's more like it, thanks @zyc9012!

zyc9012 reviewed (commented):
Thank you @Freaky 👍

Commit:
deno fmt README.md

Commit:
Fix CI for earlier node versions

Node hardcodes its default TLS certificates, meaning older versions do
not support the root certificate currently used by LetsEncrypt.

Use --use-openssl-ca to enable the use of the system's OpenSSL
certificates.

Since node v7 doesn't support NODE_OPTIONS, conditionally modify the
test script with sed.

Freaky commented:
CI failures look to be a result of https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/

Instructing it to use OpenSSL's doesn't appear to help, hmm.

zyc9012 commented:
Thank you @Freaky. I fixed the CI and squashed my commit with yours.
